src/main/java/de/tum/cit/aet/artemis/communication/web/FaqResource.java

```diff
@@ -0,0 +1,180 @@
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+import de.tum.cit.aet.artemis.core.service.AuthorizationCheckService;
+import de.tum.cit.aet.artemis.core.util.HeaderUtil;
+
+/**
+ * REST controller for managing Faqs.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class FaqResource {
+
+    private static final Logger log = LoggerFactory.getLogger(FaqResource.class);
+
+    private static final String ENTITY_NAME = "faq";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final FaqService faqService;
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public FaqResource(FaqService faqService, CourseRepository courseRepository, AuthorizationCheckService authCheckService) {
+
+        this.faqService = faqService;
+        this.courseRepository = courseRepository;
+        this.authCheckService = authCheckService;
+    }
+
+    /**
+     * POST /faqs : Create a new faq.
+     *
+     * @param faq the faq to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new faq, or with status 400 (Bad Request) if the faq has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("faqs")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Faq> createFaq(@RequestBody Faq faq) throws URISyntaxException {
+        log.debug("REST request to save Faq : {}", faq);
+        if (faq.getId() != null) {
+            throw new BadRequestAlertException("A new faq cannot already have an ID", ENTITY_NAME, "idExists");
+        }
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, faq.getCourse(), null);
+
+        Faq savedFaq = faqService.save(faq);
+        return ResponseEntity.created(new URI("/api/faqs/" + savedFaq.getId())).body(savedFaq);
+    }
+
+    /**
+     * PUT /faqs/{faqId} : Updates an existing faq.
+     *
+     * @param faq   the faq to update
+     * @param faqId id of the faq to be updated
+     * @return the ResponseEntity with status 200 (OK) and with body the updated faq, or with status 400 (Bad Request) if the faq is not valid, or with status 500 (Internal
+     *         Server Error) if the faq couldn't be updated
+     */
+    @PutMapping("faqs/{faqId}")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Faq> updateFaq(@RequestBody Faq faq, @PathVariable Long faqId) {
+        log.debug("REST request to update Faq : {}", faq);
+        if (faqId == null || !faqId.equals(faq.getId())) {
+            throw new BadRequestAlertException("Id of FAQ and path must match", ENTITY_NAME, "idNull");
+        }
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, faq.getCourse(), null);
+        Faq existingFaq = faqService.findById(faqId).orElseThrow(() -> new EntityNotFoundException("FAQ not found", faqId));
```

Comment:
You can use the default method from the repository that already does that for you with `faqRepository.findByIdElseThrow(faqId)` (also for other occurences)